src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Tests.cs

```diff
@@ -44,5 +44,16 @@ public void PriorityQueue_Generic_EnqueueDequeue_LargerThanMin()
             Assert.Equal("three", queue.Dequeue());
             Assert.Equal("four", queue.Dequeue());
         }
+
+        [Fact]
+        public void PriorityQueue_Generic_EnqueueDequeue_EqualToMin()
+        {
+            PriorityQueue<string, int> queue = SmallPriorityQueueFactory(out HashSet<(string, int)> enqueuedItems);
+
+            string actualElement = queue.EnqueueDequeue("one-not-to-enqueue", 1);
+
+            Assert.Equal("one-not-to-enqueue", actualElement);
```

Comment:
Nit: the correctness of the assertion really depends on the implementation of `SmallPriorityQueueFactory`.